vulnerable_people_form/integrations/ladcode_shielding_advice_lookup.py

```diff
@@ -0,0 +1,71 @@
+import csv
+from datetime import datetime
+import pytz
+import logging
+
+from vulnerable_people_form.form_pages.shared.logger_utils import create_log_message, log_event_names
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_NATIONAL_SHIELDING_ADVICE = 'NO'
+
+la_shielding_advice_data = {}
+
+FIELDNAMES = [
+    'ladcode',
+    'ladname',
+    'shielding_advice',
+    'start_date',
+    'start_time'
+]
+
+
+def init(shielding_advice_csv_file: str):
+    global la_shielding_advice_data
+
+    valid_applicable_records = []
+
+    with open(shielding_advice_csv_file, 'r') as f:
+        la_shielding_advice = csv.DictReader(f, delimiter=',')
+
+        # Find all non-future records
+        for la in la_shielding_advice:
+            la['start_datetime'] = _get_start_datetime(la)
+            now = datetime.utcnow().replace(tzinfo=pytz.utc)
+            if la['start_datetime'] < now:
+                valid_applicable_records.append(la)
+        # Sort by date for every ladcode in ascending order
+        sorted_advice = sorted(
+            valid_applicable_records,
+            key=lambda k: (k['ladcode'], k['start_datetime']),
+            reverse=False)
+        # Keep the latest record only.
+        for record in sorted_advice:
+            ladcode = record['ladcode']
+            la_shielding_advice_data[ladcode] = record
+
+
+def get_shielding_advice_by_ladcode(ladcode: str):
+    """
+    fetches the shielding advice status for supplied LAD Code
+    """
+    if ladcode not in la_shielding_advice_data:
+        logger.info(create_log_message(
+            log_event_names["LADCODE_NOT_IN_FILE"],
+            f"{ladcode} not in shielding advice file. Using default: {DEFAULT_NATIONAL_SHIELDING_ADVICE}"))
+        return DEFAULT_NATIONAL_SHIELDING_ADVICE
+
+    shielding_advice = la_shielding_advice_data[ladcode]["shielding_advice"]
+    logger.info(create_log_message(
+        log_event_names["SHIELDING_ADVICE_FOR_LADCODE_SUCCESS"],
+        f"{ladcode} has Shielding Advice {shielding_advice}"))
+    return shielding_advice
+
+
+def _get_start_datetime(la_shielding_advice):
+    return pytz.timezone("Europe/London").localize(

```

Comment:
I think if we control  this file we should decide to store these dates in the file as UTC datetimes and sidestep this weirdness.

Reply:
Sure things... we can do that... Though I suspect that the file we receive will be British time.... Just wondering if we should check whether we can get UTC (from source) or not ?

Reply:
I suspect they will be in London time, yes. I think we should check that and store them as UTC though.

Reply:
done !